src/Microsoft.AspNet.Cors.Core/CorsResult.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// Results returned by <see cref="ICorsEngine"/>.
+    /// </summary>
+    public class CorsResult
+    {
+        private long? _preflightMaxAge;
+
+        /// <summary>
+        /// Gets a value indicating whether the result is valid.
+        /// </summary>
+        public bool IsValid
+        {
+            get
+            {
+                return ErrorMessages.Count == 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the error messages.
+        /// </summary>
+        public IList<string> ErrorMessages { get; } = new List<string>();
```

Comment:
We should remove ErrorMessages. The only two outcomes of the CORS algorithm should be adding additional headers to a simple request (simple requests are never prevented) and a 204 response for a preflight request. An OPTIONS returning 204 without Access-Control-Allow-Origin should be interpreted as a preflight request failure by the caller.